### DISCUSSION TITLE
Basic Quesion about Setting up Libraries

### DISCUSSION THREAD

**Discussion by jprobins** [advplyr/audiobookshelf#2062] [General]
I am afraid this question is very basic and betrays my inexperience.  I have installed advplyr/audiobookshelf via Container Station (a/k/a Docker) on a QNAP NAS. I am at the point of adding my first library.  However, I do not seem to be able to get ABS to "see" the relevant directory.  The directory containing the audiobooks is on the same NAS. I have tried writing in the directory name manually. I either get a message that the name is invalid or it accepts the name and tells me (after scan) that it is empty.  I simply cannot seem to find the directory (or anything that looks familiar) using the browse function.  I expect I am doing something very simple and basic wrong. Any held or pointing towards a good resource would be much appreciated.

**Comment by nichwall** [comment]
Can you share your docker compose/mount points?

It's the part where you basically tell docker that `/path/on/computer` should be located at `/path/in/container`

If you haven't set them up, the docker container doesn't have any idea where things are on your computer.

**Comment by nhowland** [comment]
Just got ABS up and running on my QNAP and have a couple tips that helped me.

- Create a media location on the NAS and note its internal path. If you've got it on a normal volume it's likely something like "/folder/media". You can check the internal path name via the QNAP Control Panel app >> Users applet. Open it and go to the "Shared Folder" panel. In the properties of whichever volume/folder you'll find the internal path. Create as many root level media locations as needed for each library in ABS you want to create. For me I did one for audiobooks and one for ebooks.
- Create config and metadata data folders and note their internal paths. I know ABS can create these during deployment, but I had more luck doing this ahead of time.
- Once you've got all the folder locations created and their paths noted build a docker YAML file as specified on the [ABS website](https://www.audiobookshelf.org/docs#docker-compose-install).
Mine looks something like this (note: I'm running ABS on a USB flash drive so have the internal path running via the "share" aggregate folder):
```
version: "3.7"
services:
  audiobookshelf:
    image: ghcr.io/advplyr/audiobookshelf:latest
    ports:
      - 13378:80
    volumes:
      - /share/USBDRIVE/abs/audiobooks:/audiobooks
      - /share/USBDRIVE/abs/ebooks:/ebooks
      - /share/USBDRIVE/abs/config:/config
      - /share/USBDRIVE/abs/metadata:/metadata
```
- In Container Station create a new "Application". (This appears to be the only way to specify a YAML file.)
- Name the container, copy in the YAML code and select Create (I left everything else default)
- Once the ABS container is up and running you should be able to hit the main page by browsing to QNAP_IP:13378 (or whatever port you set it to in the YAML)
- Create the ABS admin account and the config/metadata mount points should be automatically used for the initial setup.
- When creating a library you should be able to use the ABS user interface to select the library mount point you set in the YAML.
- Once configured any media copied into the library folder should automatically show up in ABS

Sidenote... I initially tried the QPKG of ABS, but ended up have storage issues that were hard to troubleshoot. This method is a more configurable option.
